hastory-server-data/src/Hastory/Server/Data/Username.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Hastory.Server.Data.Username where
+
+import Data.Aeson (FromJSON(parseJSON), ToJSON(toJSON), withText)
+import qualified Data.CaseInsensitive as CI
+import Data.Proxy (Proxy(Proxy))
+import qualified Data.Text as T
+import Data.Validity
+import Database.Persist (PersistField(fromPersistValue, toPersistValue), PersistValue(PersistText))
+import Database.Persist.Sql (PersistFieldSql(sqlType))
+
+newtype Username =
+  Username
+    { unUsername :: CI.CI T.Text
```

Comment:
I'm fine with case-insensitive text, but is there a reason why you wanted it?

Reply:
I like behavior that the username "steven" and "Steven" are considered equivalent by the server. That rationale is personal though, so I can remove it if you'd prefer.

For the moment, I added a new property test to the `UsersSpec` module to make sure that the behavior is correct.

Reply:
I don't think this brings you anything tbh. It's also different than the other tools I have so it would be an extra maintenance burden.